protected/humhub/libs/StdClassConfig.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * @link      https://www.humhub.org/
+ * @copyright Copyright (c) 2023 HumHub GmbH & Co. KG
+ * @license   https://www.humhub.com/licences
+ */
+
+namespace humhub\libs;
+
+use ReflectionObject;
+
+class StdClassConfig
```

Comment:
Is this used? How does it work?

Reply:
In order to not occupy any property names, some additional data can be stored in an attached "config" object. This is accessible through the `static::config()` function.

It is now only available in the `StdClassConfigurable` extension of `StdClass`

It stores the default value (that will be returned if a non-existing property was read. It also allows to lock the property set (so no new properties can be added).